include/linux/df-detection.h

```diff
@@ -0,0 +1,35 @@
+#ifndef __LINUX_DOUBLE_FETCH_DETECTION_H__
```

Comment:
Please make this header look like existing ones (you can check include/linux/kcsan.h for an example).
First, add the SPDX license header at the top (`/* SPDX-License-Identifier: GPL-2.0 */`)
Second, mind the vertical spacing, add blank lines between logically separate parts.